btc-lsp/src/BtcLsp/Psbt/PsbtOpener.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module BtcLsp.Psbt.PsbtOpener (openChannelPsbt) where
+
+import BtcLsp.Import
+import qualified BtcLsp.Math.OnChain as Math
+import BtcLsp.Psbt.Utils
+  ( finalizePsbt,
+    fundPsbtReq,
+    openChannelReq,
+    psbtFinalizeReq,
+    psbtVerifyReq,
+    unspendUtxoLookup,
+    releaseUtxosLocks,
+    releaseUtxosPsbtLocks,
+    lockUtxos
+  )
+import qualified Data.Map as M
+import qualified LndClient as Lnd
+import qualified LndClient.Data.ChannelPoint as Lnd
+import qualified LndClient.Data.FinalizePsbt as FNP
+import qualified LndClient.Data.FundPsbt as FP
+import qualified LndClient.Data.ListUnspent as LU
+import qualified LndClient.Data.OpenChannel as Lnd
+import qualified LndClient.RPC.Katip as Lnd
+import qualified UnliftIO.STM as T
+import qualified LndClient.Data.OutPoint as OP
+import qualified UnliftIO.Exception as UE
+
+sumAmt :: [PsbtUtxo] -> MSat
+sumAmt utxos = sum $ getAmt <$> utxos
+
+autoSelectUtxos :: Env m => OnChainAddress 'Fund -> MSat -> ExceptT Failure m FP.FundPsbtResponse
+autoSelectUtxos addr amt = withLndT Lnd.fundPsbt ($ req)
+  where
+    req = fundPsbtReq $ FP.TxTemplate [] (M.fromList [(coerce addr, amt)])
+
+utxoLeaseToPsbtUtxo :: Map OP.OutPoint LU.Utxo -> FP.UtxoLease -> Maybe PsbtUtxo
+utxoLeaseToPsbtUtxo l ul = psbtUtxo . LU.amountSat <$> M.lookup op l
+  where
+    op = FP.outpoint ul
+    psbtUtxo amt = PsbtUtxo {
+      getAmt = amt,
+      getLockId = Just . UtxoLockId $ FP.id ul,
+      getOutPoint = op
+    }
+
+mapLeaseUtxosToPsbtUtxo :: Env m => [FP.UtxoLease] -> ExceptT Failure m [PsbtUtxo]
+mapLeaseUtxosToPsbtUtxo lockedUtxos = do
+  releaseUtxosLocks lockedUtxos
+  l <- unspendUtxoLookup
+  newLockedUtxos <- lockUtxos (FP.outpoint <$> lockedUtxos)
+  case sequence $ utxoLeaseToPsbtUtxo l <$> newLockedUtxos of
+    Just us -> pure us
+    Nothing -> do
+      $(logTM) DebugS $ logStr $
+        "Cannot find utxo in utxos:" <> inspect lockedUtxos <> " lookupMap: " <> inspect l
+      throwE $ FailureInternal "Cannot find utxo in unspent list"
+
+
+fundChanPsbt ::
+  (Env m) =>
+  [PsbtUtxo] ->
+  OnChainAddress 'Fund ->
+  OnChainAddress 'Gain ->
+  Money 'Lsp 'OnChain 'Gain ->
+  Money 'Usr 'OnChain 'Loss ->
+  ExceptT Failure m Lnd.Psbt
+fundChanPsbt userUtxos chanFundAddr changeAddr lspFee minerFee = do
+  let userFundingAmt = sumAmt userUtxos - coerce lspFee - coerce minerFee
```

Comment:
It should be `let userFundingAmt = sumAmt userUtxos - coerce lspFee`. At the moment it's very clear from the interface how much fees user is paying. If we include mining fee - it will become much more complex, let's not do it.